src/polyfills/custom-elements.js

```diff
@@ -491,9 +479,9 @@ class Registry {
     // I would love to not have to hold onto all of the roots, since it's a
     // memory leak. Unfortunately, there's no way to iterate a list and hold
     // onto its contents weakly.
-    this.roots_.forEach((tree) => {
+    for (const tree of this.roots_) {
```

Comment:
Does iterator here resolve the issue in the above comment at all? Not clear on what "hold onto its contents weakly" means

Reply:
No, we need a `WeakSet` here, but you can't iterate a `WeakSet`.